pkgs/native_toolchain_c/lib/src/cbuilder/cbuilder.dart

```diff
@@ -53,6 +53,24 @@ class CBuilder extends CTool implements Builder {
   CBuilder.library({
     required super.name,
     super.assetName,
+    /// The list of source files to build the library.
+    ///
+    /// This should include both C source files (e.g., `.c`) and header files
+    /// (e.g., `.h`). Including header files ensures that changes to them
+    /// invalidate the build cache, triggering recompilation when necessary.
+    /// For example, for a package named `native_add_library`:
+    /// ```dart
+    /// sources: [
+    ///   'src/native_add_library.c',
+    ///   'src/native_add_library.h',
+    ///   'src/dart_api_dl.c',
```

Comment:
Omit `dart_api_dl` from the documentation, it doesn't add extra info but it does complicate things. That's not great for documentation.